common/src/test/scala/com/gu/multimedia/storagetier/vidispine/VidispineCommunicatorSpec.scala

```diff
@@ -310,4 +313,33 @@ class VidispineCommunicatorSpec extends Specification with AfterAll with Mockito
       vx22.get.getLikelyFile must beSome
     }
   }
+
+  "VidispineCommunicator.filesByProject" should {
+    "find items associated with a project id" in {
+      val mockHttp = mock[HttpExt]
+
+      val listResponseFirst20 = HttpResponse(StatusCodes.OK, entity=HttpEntity(readSampleDoc("project_24173_contents_a.json")))
+      val listResponseNext20 = HttpResponse(StatusCodes.OK, entity=HttpEntity(readSampleDoc("project_24173_contents_b.json")))
+      val listResponseLast4 = HttpResponse(StatusCodes.OK, entity=HttpEntity(readSampleDoc("project_24173_contents_c.json")))
+      val listResponseNoHits = HttpResponse(StatusCodes.OK, entity=HttpEntity(readSampleDoc("project_24173_contents_d_no_hits.json")))
+
+      mockHttp.singleRequest(argThat((h: HttpRequest) => h.uri.path.toString().contains("first=1")), any, any, any) returns Future(listResponseFirst20)
+      mockHttp.singleRequest(argThat((h: HttpRequest) => h.uri.path.toString().contains("first=21")), any, any, any) returns Future(listResponseNext20)
+      mockHttp.singleRequest(argThat((h: HttpRequest) => h.uri.path.toString().contains("first=41")), any, any, any) returns Future(listResponseLast4)
+      mockHttp.singleRequest(argThat((h: HttpRequest) => h.uri.path.toString().contains("first=45")), any, any, any) returns Future(listResponseNoHits)
+
+      val toTest = new VidispineCommunicator(fakeConfig) {
+        override def callHttp: HttpExt = mockHttp
+      }
+
+      val result = Await.result(toTest.getFilesOfProject(23, 100), 1.second)
+      result.size mustEqual 44
+
+      val result2 = Await.result(toTest.getFilesOfProject(23, 33), 1.second)
+      result2.size mustEqual 44
+
+      val result3 = Await.result(toTest.getFilesOfProject(23, 20), 1.second)
+      result3.size mustEqual 44
+    }
```

Comment:
Are these not all the exact same call? `toTest.getFilesOfProject(23, 20)`

Reply:
Basically. I used them to verify that you could remove `|| results.size < pageSize` from `if (results.isEmpty || results.size < pageSize || start > 10000) {`.

Reply:
Cool - might as well remove the last two though in that case now they are not needed